File: snippets/fsharp/System/FuncT1,T2,T3,T4,TResult/Overview/Func5.fs
module Func5

// <Snippet2>
open System

let indexOf (s: string) s2 pos chars comparison =
    s.IndexOf(s2, pos, chars, comparison) 

let title = "The House of the Seven Gables"
let finder = Func<string, int, int, StringComparison, int>(indexOf title)
let mutable position = 0

while position > -1 do
    let characters = title.Length - position
    position <- 
        finder.Invoke("the", position, characters, StringComparison.InvariantCultureIgnoreCase)
    if position >= 0 then
        position <- position + 1
        printfn $"'The' found at position {position} in {title}."
// </Snippet2>

File: snippets/csharp/System/FuncT1,T2,T3,T4,TResult/Overview/Func5.cs
// <Snippet2>
using System;

public class FuncDelegateExample
{
    public static void Run()
    {
        string title = "The House of the Seven Gables";
        int position = 0;
        Func<string, int, int, StringComparison, int> finder = title.IndexOf;
        do
        {
            int characters = title.Length - position;
            position = finder("the", position, characters,
                            StringComparison.InvariantCultureIgnoreCase);
            if (position >= 0)
            {
                position++;
                Console.WriteLine($"'The' found at position {position} in {title}.");
            }
        } while (position > 0);
    }
}
// </Snippet2>
